readthedocs/search/parsers.py

```diff
@@ -133,28 +148,110 @@ def _parse_sections(self, title, body):
             )
             if content:
                 yield {
-                    'id': '',
-                    'title': title,
-                    'content': content,
+                    "id": "",
+                    "title": document_title,
+                    "content": content,
                 }
         except Exception as e:
-            log.info('Unable to index section', section=str(e))
+            log.info("Unable to index section", section=str(e))
 
         # Index content from h1 to h6 headers.
-        for head_level in range(1, 7):
-            tags = body.css(f'h{head_level}')
-            for tag in tags:
+        for section in [body.css(f"h{h}") for h in range(1, 7)]:
+            for tag in section:
                 try:
-                    title, id = self._parse_section_title(tag)
+                    title, _id = self._parse_section_title(tag)
                     next_tag = self._get_header_container(tag).next
                     content, _ = self._parse_section_content(next_tag, depth=2)
                     yield {
-                        'id': id,
-                        'title': title,
-                        'content': content,
+                        "id": _id,
+                        "title": title,
+                        "content": content,
                     }
-                except Exception as e:
-                    log.info('Unable to index section.', section=str(e))
+                except Exception:
+                    log.info("Unable to index section.", exc_info=True)
+
+    def _parse_dls(self, body):
+
+        # All terms in <dl>s are treated as sections.
+        # We traverse by <dl> - traversing by <dt> has shown in experiments to render a
+        # different traversal order, which could make the tests more unstable.
```

Comment:
@stsewd understandable if you think it seems strange with this approach, but at least here we know that we are just traversing 1 <dl> at a time. If we directly iterated on <dt>s with `dl > dt[id]:not([id=""])`, then I would think the ordering is even worse. I did try to swap and the order changed.. so I just left it like this.